proto/voice-info.proto

```diff
@@ -0,0 +1,175 @@
+    - Patch number shall be increased, in case of a backwards-compatible
+      bug-fix
+*/
+message Revision {
+  uint32 version = 1;
+}
+
+/* Describes a voice model's file entry */
+message FileEntry {
+  enum Compression {
+    COMPRESSION_PLAIN = 0;      // no compression
+    COMPRESSION_BROTLI = 1;
+    COMPRESSION_BZIP2 = 2;
+    COMPRESSION_GZIP = 3;
+    COMPRESSION_ZIP = 4;
+  }
+  enum Format {
+    FORMAT_PLAIN = 0;           // plain file, no archive
+    FORMAT_TAR = 1;             // tar archive
+    FORMAT_CPIO = 2;            // cpio archive
+  }
+  string path = 1;                // relative path, mandatory
+  Compression compression = 2;    // file compression, optional
+  Format format = 3;              // file format, optional
+  string md5 = 4;                 // MD5 sum of file, mandatory
+}
+
+/* Speech model language / locale info */
+message Language {
+  /* <ISO 639 language code>_<ISO 3166 country code> */
+  enum Locale {
+    /* default: any language/country) */
+    LOCALE_ANY = 0;
+    is_IS = 1;
+    /* more entries can be appended here ... */
+  }
+  Locale locale = 1;  // locale of the voice, mandatory
+  string name = 2;    // language name of the voice, mandatory
+}
+
+/* Phonemes used for acoustic model, alphabets and symbol mappings */
+message PhonemeDescription {
+  enum PhonemeType {
+    PHONEME_TYPE_INVALID = 0;
+    /* use IPA symbols, these are always distinct single symbols */
+    PHONEME_TYPE_IPA = 1;
+    /* use "extended Sampas", these are mostly made up of multiple symbols */
+    PHONEME_TYPE_XSAMPA = 2;
+  }
+  /* Mapping between Alphabet and model representation */
+  message AlphabetMapping {
+    /*
+      Mapping of all IPA/XSampa symbols to their corresponding model value,
+      as well as meta-symbols like <sp> "short pause", <pau> "pause", etc. used
+      in model training.
+    */
+    map<string, uint32> symbols = 1;
+  }
+  PhonemeType phoneme_type = 1; // phoneme type, mandatory
+  AlphabetMapping mapping = 2;  // symbol mappings, mandatory
+  bool syllabified = 3;         // "." at each syllable boundary, optional
+  bool stress_labels = 4;       // "0" or "1" after each vowel, optional
+  string word_separator = 5;    // word separator; default "-", optional
+}
+
+/* For NN models: used neural network format */
+enum FormatNN {
+  NN_FORMAT_INVALID = 0;
+  NN_FORMAT_TORCH = 1;          // plain pyTorch
+  NN_FORMAT_TORCHSCRIPT = 2;    // Torchscript
+  NN_FORMAT_TORCH_MOBILE = 3;   // Torchscript mobile
+  NN_FORMAT_ONNX = 4;           // ONNX
+  NN_FORMAT_NNAPI = 5;          // Android NNAPI
+}
+
+/* Voice acoustic model description */
+message AcousticModel {
+  /* Type of the acoustic model */
+  enum AcousticModelType {
+    ACOUSTIC_MODEL_INVALID = 0;
+    ACOUSTIC_MODEL_FESTIVAL_UNIT_SELECTION = 1;
+    ACOUSTIC_MODEL_FESTIVAL_FLITE_GRAPHEME = 2;
+    ACOUSTIC_MODEL_FASTSPEECH2 = 3;
+    ACOUSTIC_MODEL_FASTSPEECH2_ESPNET = 4;
+  }
+  
```

Comment:
I am not sure this `AcousticModelType` is necessary, and I think merging it with `FormatNN` makes more sense. It shouldn't matter whether it's our "original" Fastspeech2 model or the ESPNET version, as long as they have the same NN format.

Additionally we need to add a model type / format for ESPNET's Text2Speech compatible models ([soon to merged into tiro-tts](https://gitlab.com/tiro-is/tiro-tts/-/merge_requests/34/diffs?commit_id=6b0f5fc6e71eafc02a33a4d81ef83cc3afd3c6ac#c69697097e374c2151edf96595b9c4a08d309657_75_76)).

Reply:
There will be FLite models as well, these are not NN. And I thought, we could have e.g. `ACOUSTIC_MODEL_FASTSPEECH2` in ONNX format. The inferencing of `ACOUSTIC_MODEL_FASTSPEECH2` in like `NN_FORMAT_TORCH` format will certainly be different than `ACOUSTIC_MODEL_FASTSPEECH2_ESPNET`in `NN_FORMAT_TORCH`. Or is the usage and input parameters always the same ?